### DISCUSSION TITLE
How to resolve "Recv failure: Connection was reset"

### DISCUSSION THREAD

**Discussion by Rod-in-NM** [community/community#49456] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I started a new project, where I am following a tutorial. I created a repo in GitHub, then cloned it to my local machine. I started working on it, making commits locally. Now I'm ready to push to remote. However, I forgot to set my "user.email" property in the local repo. The repo in GitHub is public, so when I tried to push to remote it gave me an error of

> [remote rejected] main -> main (push declined due to email privacy restrictions)

So, I changed the "user.email" in my local repository. Then I tried pushing to the remote again. Now I'm getting a different error:

> fatal: unable to access 'https://github.com/Rod-in-NM/Netcore-TS.git/': Recv failure: Connection was reset

Now, I'm stumped. I don't know what to do to resolve this.

**Comment by labrinyang** [comment]
i am new to pycharm,but i learn from others:"https://blog.csdn.net/qq_38415505/article/details/83687207"
this may help you. you just need to enter the code below:
git config --global --unset http.proxy 
git config --global --unset https.proxy

**Reply by ElijahTobs** [reply]
Thank you!
This helped 👍

**Reply by Rod-at-DOH** [reply]
@2665477495 thank you for your response. However, I was looking more for how to I rewrite Git's history to replace everywhere my email address is listed, with the NoReply email address that GitHub creates for us when creating a new GitHub account.

**Reply by airtower-luna** [reply]
If you want to change an email address in already existing commits setting `user.email` is only the first step, you'll still need to change the actual commits. See [Git Tools - Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History), and if you have a lot of commits that need fixing take a look at [`git filter-repo`](https://github.com/newren/git-filter-repo).

**Reply by Tapir-Tazuli** [reply]
This worked like a charm! Thank you!

**Reply by NoneOfEver** [reply]
what can I say ?  the solution from Chinese website finally worked!!

**Reply by labrinyang** [reply]
better to enter the code below,but not one of them will 100% help:
```
git config --global --unset http.proxy 
git config --global --unset https.proxy

```

**Reply by ashioyajotham** [reply]
This worked. Cheers!

**Reply by cryptodj413** [reply]
sorry, but it doesn't work in my situation. still unsolved.

**Reply by ashioyajotham** [reply]
Hello @co-developer4 could you go and check your .gitconfig file then adjust the `postBuffer`
![image](https://github.com/community/community/assets/71454095/fc54705e-a9f4-4a0c-b00b-f9d414f0da17)

**Reply by ysh11o-2020** [reply]
U are my hero！

**Reply by TommySoi9** [reply]
That good U are my God

**Reply by afondiel** [reply]
Worked like magic for me as well. Thanks @2665477495

**Reply by z4idea** [reply]
U are my hero!

**Reply by navdotnetreqs** [reply]
> Hello @co-developer4 could you go and check your .gitconfig file then adjust the `postBuffer`

What does that even do? I had sudden issues with gitkraken and that fixed it! Thanks!

**Comment by GeekyHacks** [comment]
it doesnt resolve my issue I still receive same issue with any repository I try to clone or push/pull into my laptop; although it's working from GitHub desktop

**Reply by KhairallahA** [reply]
I think it is banned in Yemen. I had the same problem several times before but for hours and then it works normally. You can use ssh or github desktop, they're working!

**Reply by its4nas** [reply]
I also have the same problem in git bash ,but it works form github desktop. (Yemen)

**Reply by KhairallahA** [reply]
Yes, a lot of websites don't work from some days!

**Reply by mr3od** [reply]
Same here , I think the ISP is blocking certain types of http/s connections like git, curl and wget.

**Reply by Karam-Qutran** [reply]
And I say, what is the problem, and it is from Yemen Net 🤦🏻‍♂️

**Reply by wm-cortex** [reply]
Anyone in Yemen, You can use mobile 4G.

**Reply by ws-rush** [reply]
I faced same problem and solved with GitHub desktop, thanks @KhairallahA

**Reply by liu-yuxuan-C137** [reply]
maybe shutting down the VPN works？

**Reply by rvian** [reply]
for some reason turning my VPN on solved it.
maybe its something to do with my ISP "Claro" and their DNS.

**Comment by Spixmacaw** [comment]
For my project, I debugged my function that contain UnityWebRequest in the script and the problem was cause by the website you requesting from OR your internet. This is a part of my code:

IEnumerator GetESTTime()
    {
        var request = UnityWebRequest.Get("http://worldtimeapi.org/api/timezone/EST");
        yield return request.SendWebRequest();

//part of the code for debugging ----
        if(request.isHttpError || request.isNetworkError) 
        {
            Debug.Log("Error");
            //setting my player date time to default when I don't get the date from the web
            playerDateTime = DateTime.Now;
            yield break;
        }
//part of the code for debugging ----

        else
        {
        // my other functions
        }

**Comment by Spixmacaw** [comment]
Select Topic Area
Question

Body
I started a new project, where I am following a tutorial. I created a repo in GitHub, then cloned it to my local machine. I started working on it, making commits locally. Now I'm ready to push to remote. However, I forgot to set my "user.email" property in the local repo. The repo in GitHub is public, so when I tried to push to remote it gave me an error of

[remote rejected] main -> main (push declined due to email privacy restrictions)

So, I changed the "user.email" in my local repository. Then I tried pushing to the remote again. Now I'm getting a different error:

fatal: unable to access 'https://github.com/Rod-in-NM/Netcore-TS.git/': Recv failure: Connection was reset

Now, I'm stumped. I don't know what to do to resolve this.

**Comment by iCode-Cat** [comment]
git push --force solves it.

**Reply by Davisme2** [reply]
It's works! Thank you so much

**Reply by VicenteCode** [reply]
Muchas gracias, con eso resolví el error.

**Comment by luabud** [comment]
This was the only thing that worked for me: https://stackoverflow.com/questions/67230241/fatal-unable-to-access-https-dev-azure-com-xxx-openssl-ssl-connect-connec/69712045#69712045

**Reply by js-phys** [reply]
That helped for me when using git with dev.azure.com (System: Win11)!
In fact, it's a **driver problem** of the Intel WiFi card. One could simply turn off TCP/IP Version 6 in the settings. However, a better solution is to turn off the prioritization in the Driver.

- Open the intel Connectivity Performance Suite
![image](https://github.com/user-attachments/assets/2fe37848-91dc-44ff-b140-7adb1fd6bc5c)

- Go to Advanced Settings and uncheck the Prioritization
![image](https://github.com/user-attachments/assets/a5ac4ed6-dd16-4f7b-b55e-7847ee6af596)

Now fetch, push, pull as always.

**Reply by Rod-at-DOH** [reply]
@js-phys how do I get to that Intel Connectivity Performance Suite? I tried entering that at my Windows 11 search prompt, but all it did was bring me to documentation on it.

**Reply by js-phys** [reply]
> @js-phys how do I get to that Intel Connectivity Performance Suite? I tried entering that at my Windows 11 search prompt, but all it did was bring me to documentation on it.

I think I installed it via the [Intel® Driver & Support Assistant](https://www.intel.com/content/www/us/en/support/intel-driver-support-assistant.html). Otherwise one could maybe install it directly [Intel® Connectivity Performance Suite](https://www.intel.com/content/www/us/en/download/738623/838299/intel-connectivity-performance-suite-icps-for-intel-wireless-products-and-intel-connectivity-manager-icm.html).

**Reply by cnt221** [reply]
@js-phys , I focussed on your first line, **That helped for me when using git with dev.azure.com (System: Win11)!** and flicked the Prioritization switch. Visual Studio was already open showing the Recv failure:, I clicked Push and the commit came to life, kudos to you! This would have taken me a couple of days to have figured this out. Thank you!

**Reply by mkillow** [reply]
@js-phys I love you, thank you.  disabling prioritization instantly fixed my problems.   Clearly intel doesn't think GIT traffic is important at all.

**Comment by allenhongzhou** [comment]
If you use VPN to have access to GitHub and you're windows users, then you need to set you git proxy config same as your VPN ip and VPN port, you can refer to https://blog.csdn.net/weixin_44223180/article/details/133059575

**Reply by YaoJianyu77** [reply]
多谢

**Reply by jessiestang** [reply]
Thanks!

**Comment by liumingda** [comment]
try git config --global http.proxy http://127.0.0.1:7890
     git config --global https.proxy http://127.0.0.1:7890                                                                                                                                          
 then open VPN ; git clone...

**Reply by Jeffjeno** [reply]
You're my  hero, bro!

**Comment by syedlabib4** [comment]
Here’s what’s happening and how to fix it:

Email privacy restriction:

GitHub rejects commits if your user.email is private and the commit was made with a different email.

To fix this, update the commits with the correct email using
git commit --amend --author="Your Name <your-email@example.com>"
or for multiple commits, use git rebase -i and git commit --amend on each.

Connection reset after changing email:

This is usually network-related, not GitHub.

Check your internet connection, VPN, or firewall.

Alternatively, try pushing over SSH instead of HTTPS:
git remote set-url origin git@github.com:username/repo.git
git push -u origin main

**Comment by raybeneth** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我最近正在休假中，无法亲自回复你的邮件。
